perf_event_output needn't call CreateGetCpuId()

Birch-san opened the pull request:
Resolves https://github.com/iovisor/bpftrace/issues/1375

`IRBuilderBPF::CreatePerfEventOutput` was emitting a `bpf_raw_smp_processor_id()` call, to lookup the ID of the current CPU. But there's a cheaper way to do this -- set the `BPF_F_CURRENT_CPU` flag. The kernel will call `smp_processor_id()` for us.

##### Checklist

- [x] Language changes are updated in `docs/reference_guide.md`
  - no language changes
- [x] User-visible and non-trivial changes updated in `CHANGELOG.md`
  - no user-visible changes. change is trivial.
- [x] The new behaviour is covered by tests
  - no language changes

fbs commented:
Ack but can you put the info from your cover letter (the 2nd paragraph) in the first commit message (maybe squash the other two in as well?). We'd like history to be useful outside of github

Birch-san commented:
@fbs I've now squashed the commits together, and included in the commit message the description of the PR.

LLVM 12 pipeline having some trouble building Docker container, but I think that's unrelated to my changes (it was passing before I squashed the commits).
